File: src/main/kotlin/me/racci/bloodnight/specialmobs/mobs/witch/ThunderWizard.java
package me.racci.bloodnight.specialmobs.mobs.witch;

import de.eldoria.bloodnight.specialmobs.SpecialMobUtil;
import org.bukkit.Particle;
import org.bukkit.entity.Witch;
import org.bukkit.event.entity.ProjectileLaunchEvent;

public class ThunderWizard extends AbstractWitch {
    public ThunderWizard(Witch witch) {
        super(witch);
    }

    @Override
    public void tick() {
        SpecialMobUtil.spawnParticlesAround(getBaseEntity(), Particle.SPELL_INSTANT, 15);
        if (canShoot(4) && getBaseEntity().getTarget() != null) {
            getBaseEntity().getLocation().getWorld().strikeLightning(getBaseEntity().getTarget().getLocation());
            shot();
        }
    }

    @Override
    public void onProjectileShoot(ProjectileLaunchEvent event) {
        event.setCancelled(true);
    }
}

File: src/main/kotlin/me/racci/bloodnight/specialmobs/mobs/witch/ThunderWizard.kt
package me.racci.bloodnight.specialmobs.mobs.witch

import me.racci.bloodnight.specialmobs.SpecialMobUtil
import me.racci.bloodnight.specialmobs.mobs.abstractmobs.AbstractWitch
import org.bukkit.Particle
import org.bukkit.entity.Witch
import org.bukkit.event.entity.ProjectileLaunchEvent

class ThunderWizard(witch: Witch) : AbstractWitch(witch) {

    override fun tick() {
        SpecialMobUtil.spawnParticlesAround(baseEntity, Particle.SPELL_INSTANT, 15)
        if (canShoot(4) && baseEntity.target != null) {
            baseEntity.location.world.strikeLightning(baseEntity.target!!.location)
            shot()
        }
    }

    override fun onProjectileShoot(event: ProjectileLaunchEvent) {
        event.isCancelled = true
    }
}